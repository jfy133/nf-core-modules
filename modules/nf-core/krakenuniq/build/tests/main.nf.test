// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test krakenuniq/build
nextflow_process {

    name "Test Process KRAKENUNIQ_BUILD"
    script "../main.nf"
    process "KRAKENUNIQ_BUILD"
    config './nextflow.config'

    tag "modules"
    tag "modules_nfcore"
    tag "krakenuniq"
    tag "krakenuniq/build"
    tag "krakenuniq/download"

    setup {
        run("KRAKENUNIQ_DOWNLOAD"){
            script "../../download/main.nf"
            process {
            """
            input[0] = 'taxonomy'
            """
            }
        }
    }

    test("sarscov2 - fasta") {

        when {
            process {
                """
                ch_fastas = Channel.fromPath(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ch_seqid2mapid = Channel.fromPath(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/seqid2taxid.map', checkIfExists: true)

                ch_input = ch_fastas
                            .combine(KRAKENUNIQ_DOWNLOAD.out.output)
                            .combine(ch_seqid2mapid)
                            .map {
                                fna, tax, map ->

                                [ [id:'test'] , fna, tax, map ]
                            }

                input[0] = ch_input
                """
            }
        }

        then {
            def stablefiles = []
            file(process.out.db.get(0).get(1)).eachFileRecurse{ file -> if (!["database-build.log", "database.report.tsv", "timestamp", "taxonomy", "taxdump.tar.gz", "names.dmp", "nodes.dmp"].find {file.toString().endsWith(it)}) {stablefiles.add(file)} }
            assertAll(
                { assert process.success },
                { assert snapshot(
                        stablefiles
                    ).match()
                }
            )
        }

    }

    test("sarscov2 - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [[id:'test'], [] , [] , []]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                )
        }

    }

}
