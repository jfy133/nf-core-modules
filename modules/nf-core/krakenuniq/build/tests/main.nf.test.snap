{
    "sarscov2 - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,b7fb6c70957b2e6b6b810757928d0325"
                ],
                "db": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b7fb6c70957b2e6b6b810757928d0325"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-01-16T14:19:28.193479362"
    },
    "sarscov2 - fasta": {
        "content": [
            [
                [
                    "genome.fasta:md5,6e9fe4042a72f2345f644f239272b7e6"
                ],
                "genome.fasta:md5,6e9fe4042a72f2345f644f239272b7e6",
                "taxDB:md5,94786387249d25c99370e9ae2f50c2e1",
                "database.jdb:md5,48db5f6bb9b9e9db7f5bbdd0d8bec4a8",
                "database.kdb.counts:md5,a82149480a435210ec1e870f06b6cdb3",
                "database.kdb:md5,46542be19979e867075f9e455a314e00",
                "database.kraken.tsv:md5,be9bd0cb3a8bea30989e65aa42a73727",
                "database0.kdb:md5,46542be19979e867075f9e455a314e00",
                "seqid2taxid.map:md5,fe738f2333d57dde0db32239c04a93b8",
                "database.idx:md5,f5298aa4215c956d2f737b5988721a24",
                "library-files.txt:md5,8a9817d2de8465eb436aa3aa8696a717"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-01-16T14:33:40.951331581"
    }
}